src/components/Registries/RegistriesCompanies/index.js

```diff
@@ -13,34 +14,54 @@ import styles from './styles.module.scss';
 import { RenderRegistryItemDetails } from '../../../utils/registryFormBuilder';
 import DeleteCompanyModal from '../../Modals/DeleteCompanyModal';
 import router from '../../../router';
+import { generatePdf } from '../../../api/middleware';
 import stylesPage from '../../../utils/pagesBase.module.scss';
 
 function InvalidCompany({ id }) {
   return (
     <Card title={`ID: ${id}`} className={styles.companyCardContainer}>
       <div className={styles.companyContentContainer}>
-        This company's registry data is corrupted. Please contact
+        This company&apos;s registry data is corrupted. Please contact
         administration.
       </div>
     </Card>
   );
 }
 
+InvalidCompany.propTypes = {
+  id: PropTypes.string.isRequired,
+};
+
 function RegistriesCompanies() {
   const [expandedDetailsForCompany, setExpandedDetailsForCompany] = useState(null);
   const [isDeleteCompanyModalOpen, setIsDeleteCompanyModalOpen] = useState(false);
   const dispatch = useDispatch();
   const userWalletAddress = useSelector(
     blockchainSelectors.userWalletAddressSelector,
   );
-
+  const blockNumber = useSelector(blockchainSelectors.blockNumber);
   useEffect(() => {
     dispatch(
       registriesActions.getOfficialUserRegistryEntries.call(userWalletAddress),
     );
   }, [dispatch, userWalletAddress]);
   const registries = useSelector(registriesSelectors.registries);
 
+  const handleGenerateButton = async (companyId) => {
+    const pathName = 'certificate';
+    const blob = await generatePdf(userWalletAddress, companyId, pathName, blockNumber);
+
+    const href = URL.createObjectURL(blob);
+    const file = `${pathName}.pdf`;
+    const link = document.createElement('a');
+    link.href = href;
+    link.setAttribute('download', file);
+    document.body.appendChild(link);
+    link.click();
+    document.body.removeChild(link);
+    URL.revokeObjectURL(href);
```

Comment:
what is this and why is it like this

Reply:
What's wrong with it?

get blob from middleware,
creates a unique URL for the Blob, 
create new element - anchor for the download link,
set href as our url blob,
add download attribute for browser, and set name of download file,
add to dom,
initiating the download process,
clear code